File: 2021/day2/day2_part2.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strconv"
	"strings"
)

func main() {
	file, err := os.Open("input.txt")

	if err != nil {
		log.Fatalf("failed opening file: %s", err)
	}

	scanner := bufio.NewScanner(file)
	scanner.Split(bufio.ScanLines)
	var txtlines []string

	for scanner.Scan() {
		txtlines = append(txtlines, scanner.Text())
	}

	file.Close()

	h_pos := 0
	depth := 0
	aim := 0

	for _, line := range txtlines {
		res := strings.Split(line, " ")
		command := res[0]
		counter, _ := strconv.Atoi(res[1])
		switch {
		case command == "forward":
			h_pos = h_pos + counter
			depth = depth + (aim * counter)
		case command == "down":
			aim = aim + counter
		default:
			aim = aim - counter
		}
	}
	fmt.Println("horizontal_pos: ", h_pos)
	fmt.Println("depth: ", depth)
	fmt.Println("result: ", h_pos*depth)
}

File: 2022/day2/day2_part2.py
def main():
    file = open('input.txt', 'r')
    lines = file.readlines()

    OUTCOME = {
        "A X": 0 + 3,
        "A Y": 3 + 1,
        "A Z": 6 + 2,

        "B X": 0 + 1,
        "B Y": 3 + 2,
        "B Z": 6 + 3,

        "C X": 0 + 2,
        "C Y": 3 + 3,
        "C Z": 6 + 1
    }

    score = 0

    for line in lines:
        line = line.strip()
        if line == "":
            continue
        score += OUTCOME[line]

    print("Score:", score)

if __name__ == "__main__":
    main()